message_definitions/v1.0/ardupilotmega.xml

```diff
@@ -93,37 +93,37 @@
-        <param index="2">MagInclinationDegrees.</param>
-        <param index="3">MagIntensityMilliGauss.</param>
-        <param index="4">YawDegrees.</param>
+        <param index="1" label="declination" units="deg">MagDeclinationDegrees.</param>
+        <param index="2" label="inclination" units="deg">MagInclinationDegrees.</param>
+        <param index="3" label="intensity" units="mgauss">MagIntensityMilliGauss.</param>
+        <param index="4" label="yaw" units="deg">YawDegrees.</param>
         <param index="5">Empty.</param>
         <param index="6">Empty.</param>
         <param index="7">Empty.</param>
       </entry>
       <entry value="42005" name="MAV_CMD_FIXED_MAG_CAL_FIELD">
-        <description>Magnetometer calibration based on fixed expected field values in milliGauss.</description>
-        <param index="1">FieldX.</param>
-        <param index="2">FieldY.</param>
-        <param index="3">FieldZ.</param>
+        <description>Magnetometer calibration based on fixed expected field values.</description>
+        <param index="1" label="field X" units="mgauss">FieldX.</param>
+        <param index="2" label="field Y" units="mgauss">FieldY.</param>
+        <param index="3" label="field Z" units="mgauss">FieldZ.</param>
         <param index="4">Empty.</param>
         <param index="5">Empty.</param>
         <param index="6">Empty.</param>
         <param index="7">Empty.</param>
       </entry>
       <entry value="42424" name="MAV_CMD_DO_START_MAG_CAL">
         <description>Initiate a magnetometer calibration.</description>
-        <param index="1">uint8_t bitmask of magnetometers (0 means all).</param>
-        <param index="2">Automatically retry on failure (0=no retry, 1=retry).</param>
-        <param index="3">Save without user input (0=require input, 1=autosave).</param>
-        <param index="4">Delay (seconds).</param>
-        <param index="5">Autoreboot (0=user reboot, 1=autoreboot).</param>
+        <param index="1" label="bitmask" increment="1">uint8_t bitmask of magnetometers (0 means all).</param>
+        <param index="2" label="retry on failure" minValue="0" maxValue="1" increment="1">Automatically retry on failure (0=no retry, 1=retry).</param>
+        <param index="3" label="autosave" minValue="0" maxValue="1" increment="1">Save without user input (0=require input, 1=autosave).</param>
+        <param index="4" label="delay" units="s">Delay.</param>
+        <param index="5" label="autoreboot" minValue="0" maxValue="1" increment="1">Autoreboot (0=user reboot, 1=autoreboot).</param>
         <param index="6">Empty.</param>
         <param index="7">Empty.</param>
       </entry>
       <entry value="42425" name="MAV_CMD_DO_ACCEPT_MAG_CAL">
         <description>Initiate a magnetometer calibration.</description>
-        <param index="1">uint8_t bitmask of magnetometers (0 means all).</param>
+        <param index="1" label="bitmask" increment="1">uint8_t bitmask of magnetometers (0 means all).</param>
```

Comment:
Interesting - not seen a bitmask in a param before. You could set min an maxvalue on this.

Reply:
Improved